docs/CreateEdit.md

```diff
@@ -928,6 +944,141 @@ export const UserEdit = props => {
+
+                            <Box display="flex">
+                                <Box flex={1} mr="0.5em">
+                                    <TextInput source="first_name" resource="customers" fullWidth />
+                                </Box>
+                                <Box flex={1} ml="0.5em">
+                                    <TextInput source="last_name" resource="customers" fullWidth />
+                                </Box>
+                            </Box>
+                            <TextInput source="email" resource="customers" type="email" fullWidth />
+                            <DateInput source="birthday" resource="customers" />
+                            <Box mt="1em" />
+
+                            <Typography variant="h6" gutterBottom>Address</Typography>
+
+                            <TextInput resource="customers" source="address" multiline fullWidth />
+                            <Box display="flex">
+                                <Box flex={1} mr="0.5em">
+                                    <TextInput source="zipcode" resource="customers" fullWidth />
+                                </Box>
+                                <Box flex={2} ml="0.5em">
+                                    <TextInput source="city" resource="customers" fullWidth />
+                                </Box>
+                            </Box>
+                        </Box>
+
+                        <Box flex={1} ml="1em">
+                            
+                            <Typography variant="h6" gutterBottom>Stats</Typography>
+
+                            <SelectArrayInput source="groups" resource="customers" choices={segments} fullWidth />
+                            <NullableBooleanInput source="has_newsletter" resource="customers" />
+                        </Box>
+
+                    </Box>
+                </Box>
+                <Toolbar>
+                    <Box display="flex" justifyContent="space-between" width="100%">
+                        <SaveButton
+                            saving={formProps.saving}
+                            handleSubmitSithRedirect={formProps.handleSubmitSithRedirect}
+                        />
+                        <DeleteButton record={formProps.record} />
+                    </Box>
+                </Toolbar>
+            </form>
+        )}
+    />
+);
+```
+
+This custom form layout component uses the `FormWithRedirect` component, which wraps react-final-form's `Form` component to handle redirection logic. It also uses react-admin's `<SaveButton>` and a `<DeleteButton>`.
+
+**Tip**: When `Input` components have a `resource` prop, they use it to determine the input label. `<SimpleForm>` and `<TabbedForm>` inject this `resource` prop to `Input` components automatically. When you use a custom form layout, pass the `resource` prop manually - unless the `Input` has a `label` prop.
```

Comment:
We really should use react Context instead :)

Reply:
I know, v4 ^^

Reply:
indeed ^^